package.json

```diff
@@ -21,17 +26,20 @@
     "fs-extra": "^8.1.0",
     "lodash": "^4.17.4",
     "mkdirp": "^0.5.1",
-    "mocha": "^6.2.2",
-    "terser": "^4.4.2"
+    "mocha": "^7.0.1",
+    "terser": "^4.4.2",
+    "ts-mocha": "^6.0.0",
+    "tslint": "^6.0.0",
+    "typescript": "^3.7.5"
   },
   "scripts": {
     "benchmark": "babel-node --presets @babel/preset-env benchmark.js",
     "build": "node ./scripts/build.js",
     "check": "npm test && npm run lint && npm run build",
-    "dist": "babel ./src --out-dir dist --presets @babel/preset-env",
-    "lint": "eslint -c eslint.config.js ./src ./test",
+    "dist": "babel ./src --out-dir dist --presets @babel/preset-env && cp src/baobab.d.ts dist/.",
+    "lint": "eslint -c eslint.config.js ./src ./test && tslint src/baobab.d.ts test/suites/*.ts",
     "prepublish": "npm run check && npm run dist",
-    "test": "mocha -R spec --require ./test/register.js ./test/endpoint.js"
+    "test": "ts-mocha --reporter spec --require test/register.js test/suites/*.ts"
```

Comment:
Just found [an issue in ts-mocha](https://github.com/piotrwitek/ts-mocha/issues/49), that causes the test to pass, despite TypeScript code errors in test files. I ensured it was all good with TypeScript integration in my IDE, but that's all, test can run anyway even when there are errors...

This should be solved before merging I think.